pkg/storage/watch_cache.go

```diff
@@ -266,7 +266,7 @@ func (w *watchCache) GetAllEventsSinceThreadUnsafe(resourceVersion uint64) ([]wa
 		return result, nil
 	}
 	if resourceVersion < oldest {
-		return nil, errors.NewInternalError(fmt.Errorf("too old resource version: %d (%d)", resourceVersion, oldest))
+		return nil, errors.NewBadRequest(fmt.Sprintf("too old resource version: %d (%d)", resourceVersion, oldest))
```

Comment:
To be honest - I'm not convinced about this change.
From the logic point of view - it's not a bad request - it's totally valid request.
It's the implementation limitation, that we cannot handle this request, because we don't have old enough data.

So for me it look more like "InternalError" - the request is correct - we just aren't good enough to serve it :)